protocol/contracts/templegold/TempleGold.sol

```diff
@@ -0,0 +1,387 @@
+ * On mint, Temple Gold is distributed to DaiGoldAuction, Staking contracts and team multisig using distribution share parameters percentages set at `DistributionParams`. 
+ * Users can get Temple Gold by staking Temple for Temple Gold rewards on the staking contract or in auctions.
+ * Holders can transfer their Temple Gold to same holder address across chains.
+ * The intended owner of Temple Gold is the TempleGoldAdmin contract for admin functions. 
+ * This is done to avoid manually importing lz contracts and overriding `Ownable` with `TempleElevatedAccess`
+ */
+ contract TempleGold is ITempleGold, OFT {
+    using OFTMsgCodec for bytes;
+    using OFTMsgCodec for bytes32;
+
+    /// @notice These addresses are mutable to allow change/upgrade.
+    /// @notice Staking contract
+    ITempleGoldStaking public override staking;
+    /// @notice Dai Gold auction contract
+    IDaiGoldAuction public override daiGoldAuction;
+    /// @notice Multisig gnosis address
+    address public override teamGnosis;
+
+    /// @notice Last block timestamp Temple Gold was minted
+    uint32 public override lastMintTimestamp;
+
+    //// @notice Distribution as a percentage of 100
+    uint256 public constant DISTRIBUTION_DIVISOR = 100 ether;
+    /// @notice 1B max circulating supply
+    uint256 public constant MAX_CIRCULATING_SUPPLY = 1_000_000_000 ether; // 1B
+    /// @notice Minimum Temple Gold minted per call to mint
+    uint256 public constant MINIMUM_MINT = 10_000 ether;
+
+    /// @notice Mint chain id
+    uint128 private immutable _mintChainId;
+    uint128 private immutable _mintChainLzEid;
+
+    /// @notice Total distribtued to track total supply
+    uint256 private _totalDistributed;
+    /// @notice Total TGLD burned from all spice auctions
+    uint256 private _totalBurnedFromSpiceAuctions;
+    uint256 private _circulatingSupply;
+
+    /// @notice Whitelisted addresses for transferrability
+    mapping(address => bool) public override authorized;
+    /// @notice Distribution parameters. Minted share percentages for staking, dai gold auction and gnosis. Adds up to 100%
+    DistributionParams private distributionParams;
+    /// @notice Vesting factor determines rate of mint
+    // This represents the vesting factor
+    // If mint every 1 second instead of 1 week, vestingFactor == 35 / 604,800. It is possible but unlikely vesting factor is changed in future
+    VestingFactor private vestingFactor;
+
+    constructor(
+        InitArgs memory _initArgs
+    ) OFT(_initArgs.name, _initArgs.symbol, _initArgs.layerZeroEndpoint, _initArgs.executor) Ownable(_initArgs.executor){
+       _mintChainId = _initArgs.mintChainId;
+       _mintChainLzEid = _initArgs.mintChainLzEid;
+    }
+
+    /**
+     * @notice Set staking proxy contract address
+     * @param _staking Staking proxy contract
+     */
+    function setStaking(address _staking) external override onlyOwner {
```

Comment:
mint to staking contract would fail anyways, but why not:

```suggestion
    function setStaking(address _staking) external override onlyOwner onlyArbitrum {
```